client/src/components/History/CurrentHistory/HistoryOperations/SelectionOperations.vue

```diff
@@ -304,6 +313,29 @@ export default {
         async buildDatasetList() {
             await this.buildNewCollection("list");
         },
+        async buildDatasetListAll() {
+            let allContents = [];
+            const filterQuery = `v=dev&q=visible&qv=true&q=deleted&qv=false&q=purged&qv=false`;
+            const uri = `${getAppRoot()}api/histories/${this.history.id}/contents/datasets?${filterQuery}`;
```

Comment:
To honor the actual search filters you probably want to use something like this to translate the searchText into actual query filters:

```suggestion
            const filterQuery = `v=dev&${HistoryFilters.getQueryString(this.filterText)}}`;
            const uri = `${getAppRoot()}api/histories/${this.history.id}/contents/datasets?v=dev&${filterQuery}`;
```

That said, I would move this whole query to its own function in `client/src/components/History/model/queries.js`. That would make it easier to replace in the future and you can handle it more consistently with other queries.